tools/scripts/check_formatting.sh

```diff
@@ -27,7 +27,7 @@ fail() {
 }
 
 hash git || fail "git not found"
-hash clang-format-10 || fail "clang-format-10 not found"
+hash clang-format-12 || fail "clang-format-12 not found"
```

Comment:
This is again very ubuntu/debian specific. Here we could do the same trick like in the script above.

```sh
## when it comes first clang-format-12 has precedence if it exists
if [[ ! -z $(command -v clang-format-12 2>/dev/null) ]]; then 
    CLANG_FORMAT_CMD=$(command -v clang-format-12)
elif [[ ! -z $(command -v clang-format 2>/dev/null) ]]; then
    CLANG_FORMAT_CMD=$(command -v clang-format)
else
    echo clang-format is not installed
    exit
fi
```

To ensure the correct version we can do again a sed command like in the script above

```
CLANG_FORMAT_MAJOR_VERSION=$(${CLANG_FORMAT} --version | sed -n "s/clang-format version \([0-9]*\)\.[0-9]*\.[0-9]*/\1/p")

if [[ $CLANG_FORMAT_MAJOR_VERSION -lt 12 ]]; then
    echo clang-format major version must be 12 or greater
    exit
fi
```